common/tasks/letsencrypt.yaml

```diff
@@ -108,42 +129,209 @@
+    name: snapd
+    state: present
+
+- name: Enable snapd socket
+  ansible.builtin.systemd_service:
+    name: snapd.socket
+    enabled: true
+
+- name: Enable classic snap
+  ansible.builtin.file:
+    src: /var/lib/snapd/snap
+    dest: /snap
+    state: link
+
+- name: Ensure snapd socket started
+  ansible.builtin.systemd_service:
+    name: snapd.socket
+    state: started
+
+- name: Restart snapd service
+  ansible.builtin.systemd_service:
+    name: snapd.service
+    state: restarted
+
+- name: Restart snapd seeded service
+  ansible.builtin.systemd_service:
+    name: snapd.seeded.service
+    state: restarted
+
+- name: Install certbot
+  community.general.snap:
+    classic: true
+    name: certbot
+    state: present
+
+- name: Prepare certbot command
+  ansible.builtin.file:
+    src: /snap/bin/certbot
+    dest: /usr/bin/certbot
+    state: link
+# --- end certbot install ---
+
+- name: Register if each virtual_host certificate store exists
+  ansible.builtin.stat:
+    path: "/etc/letsencrypt/live/{{ item.virtualhost }}"
+  loop: "{{ i_virtualhosts }}"
+  register: i_virtualhosts_certificates_store
+
+- name: Populate service facts
+  ansible.builtin.service_facts:
+
+- name: Stop httpd service if present
+  ansible.builtin.systemd_service:
+    name: httpd.service
+    state: stopped
+  when: "'httpd' in services"
+
+- name: Stop HAProxy service if present
+  ansible.builtin.systemd_service:
+    name: HAProxy.service
+    state: stopped
+  when: "'HAProxy' in services"
+
+# get firewall status (save if http,https disabled)
+- name: Retrieve firewall-cmd stat
+  ansible.builtin.stat:
+    path: /usr/bin/firewall-cmd
+  register: firewall_present
+
+- name: Set firewall_exists to true if firewall-cmd is present
+  ansible.builtin.set_fact:
+    firewall_present: "{{ firewall_present.stat.exists }}"
+
+- name: Get firewall status
+  ansible.posix.firewalld_info:
+    zones: public
+  register: firewall_status
+  when: firewall_present
+
+- name: Register if http service disabled in firewalld
+  ansible.builtin.set_fact:
+    http_disabled: true
+  when:
+    - firewall_present
+    - "'http' not in firewall_status.firewalld_info.zones.public.services"
+    - http_disabled is undefined
+
+- name: Register if https service disabled in firewalld
+  ansible.builtin.set_fact:
+    https_disabled: true
+  when:
+    - firewall_present
+    - "'https' not in firewall_status.firewalld_info.zones.public.services"
+    - https_disabled is undefined
+
+- name: Enable http in firewalld
+  ansible.posix.firewalld:
+    service: http
+    zone: public
+    state: enabled
+  when: firewall_present
+
+- name: Enable https in firewalld
+  ansible.posix.firewalld:
+    service: https
+    zone: public
+    state: enabled
+  when: firewall_present
+
+- name: Set up new certificates for virtualhosts
+  ansible.builtin.command: certbot -n --standalone -d {{ item.item.virtualhost }} --agree-tos --email {{ admin_email }} certonly
```

Comment:
May not need to disable haproxy or httpd when running the certbot command with --apache

Reply:
csv2-ci which shares this task does not use apache